R/reportLCOE.R

```diff
@@ -237,12 +239,28 @@ reportLCOE <- function(gdx, output.type = "both"){
 
  # 2. sub-part: fuel cost ----
 
- # fuel cost = PE price * PE demand of technology
+ # 2.1 primary fuel cost = PE price * PE demand of technology
 
  te_annual_fuel_cost <- new.magpie(getRegions(te_inv_annuity),ttot_from2005,magclass::getNames(te_inv_annuity), fill=0)
  te_annual_fuel_cost[,,pe2se$all_te] <- setNames(1e+12 * qm_pebal[,ttot_from2005,pe2se$all_enty] / qm_budget[,ttot_from2005,] *
            setNames(vm_demPe[,,pe2se$all_te], pe2se$all_enty), pe2se$all_te)
 
+ # 2.2 second fuel cost - !only CCSinje for now! 
+    # pm_prodCouple: Second fuel production or demand per unit output of technology. Negative values mean own consumption, positive values mean coupled product. 
+      pm_prodCouple <- readGDX(gdx, "pm_prodCouple", restore_zeros = F)
+    # potential fuels and prices: not needed in this detail here, but copying approach in marginal section
+      fuels <- c("peoil","pegas","pecoal","peur", "pebiolc" , "pebios","pebioil",
+                 "seel","seliqbio", "seliqfos","seliqsyn", "sesobio","sesofos","seh2","segabio" ,
+                 "segafos","segasyn","sehe")
+      pm_PEPrice <- readGDX(gdx, "pm_PEPrice", restore_zeros = F) 
+      pm_SEPrice <- readGDX(gdx, "pm_SEPrice", restore_zeros = F)
+      Fuel.Price <- mbind(pm_PEPrice,pm_SEPrice )[,,fuels]*1e12 # convert from trUSD2005/TWa to USD2005/TWa
+
+    # calculate secondary Fuel cost for ccsinje 
+      te_annual_secFuel_cost <- new.magpie(getRegions(te_inv_annuity),ttot_from2005, "ccsinje" , fill=0)  
+      te_annual_secFuel_cost[,,"ccsinje"] <- setNames(-pm_prodCouple[,,"ccsinje"] * Fuel.Price[,,"seel"] * vm_co2CCS[,,"ccsinje.1"], "ccsinje") 
```

Comment:
I am probably missing something, but why negative here? (-pm_prodCouple)

Reply:
It says in line 249 that negative values mean own consumption, positive values mean coupled product. As I want positive cost, I need to multiply. 
I can adjust the line with the units if you want to have a more detailed documentation.